analyzers/src/SonarAnalyzer.Shared/LoggingFrameworkIdentifier.cs

```diff
@@ -0,0 +1,82 @@
+namespace Roslyn.Utilities.SonarAnalyzer.Shared;
+
+public static class LoggingFrameworkIdentifier
+{
+    public static readonly HashSet<string> MicrosoftExtensionsLoggingMethods =
+    [
+        "Log",
+        "LogCritical",
+        "LogDebug",
+        "LogError",
+        "LogInformation",
+        "LogTrace",
+        "LogWarning"
+    ];
+
+    public static readonly HashSet<string> CastleCoreOrCommonCoreLoggingMethods =
+    [
+        "Debug",
+        "DebugFormat",
+        "Error",
+        "ErrorFormat",
+        "Fatal",
+        "FatalFormat",
+        "Info",
+        "InfoFormat",
+        "Trace",
+        "TraceFormat",
+        "Warn",
+        "WarnFormat"
+    ];
+
+    public static readonly HashSet<string> Log4NetLoggingMethods =
+    [
+        "Debug",
+        "Error",
+        "Fatal",
+        "Info",
+        "Warn"
+    ];
+
+    public static readonly HashSet<string> Log4NetLoggingExtensionMethods =
+    [
+        "DebugExt",
+        "ErrorExt",
+        "FatalExt",
+        "InfoExt",
+        "WarnExt"
+    ];
+
+    public static readonly HashSet<string> NLogLoggingMethods =
+    [
+        "Debug",
+        "ConditionalDebug",
+        "Error",
+        "Fatal",
+        "Info",
+        "Trace",
+        "ConditionalTrace",
+        "Warn"
+    ];
+
+    public static readonly HashSet<string> NLogILoggerBaseLoggingMethods = ["Log"];
+
+    public static bool IsLoggingInvocation(InvocationExpressionSyntax invocation, SemanticModel model) =>
```

Comment:
Is there any other rule that will use this method except the one you already implemented?
For example, the template rules cannot use it, since they don't care about CastleCore and Log4net.

Reply:
dropped